jmxtrans-core/src/main/java/com/googlecode/jmxtrans/JmxTransformer.java

```diff
@@ -317,14 +318,24 @@ public void executeStandalone(JmxProcess process) throws Exception {
 	/**
 	 * Processes files into Server objects and then processesServers into jobs
 	 */
-	private void startupSystem() throws LifecycleException {
-		// process all the json files into Server objects
+	private void startupSystem() throws Exception {
 		this.processFilesIntoServers();
-
-		// process the servers into jobs
+		this.initializeExecutors();
+		this.registerMBeans();
 		this.processServersIntoJobs();
 	}
 
+	private void initializeExecutors() throws MalformedObjectNameException {
+		this.initializeExecutors(queryExecutorRepository);
+		this.initializeExecutors(resultExecutorRepository);
+	}
+
+	private void initializeExecutors(ExecutorRepository executorRepository) throws MalformedObjectNameException {
+		for (Server server : this.masterServersList) {
```

Comment:
I'm wondering if it wouldn't make sense to make the ExecutorRepository lazy initialized (on a call to `getExecutor()`). This would allow to not have initialization logic outside of the Guava modules.

Reply:
I thought JmxTransformer class has initialization responsobility, isn't it?

Reply:
It does have some initialization responsibilities left, but I'm trying to slowly remove what's left when it make sense. I'm not sure it is worth it in this case...

Reply:
I've tried to implement it in JmxModule, but I need server list. And server list initialization occurs over here.
I thought about extracting ServerRepository and inject it in ExecutorRepository but it seems more complicated to understand.

Reply:
Understood. Not worth the effort. Let's move on!  :+1: